### PR TITLE
WebApp: Adjusted app settings implementation (enabled custom)

### PR DESCRIPTION
**Pull request by MrMCake**
# Change

- Enabled custom app settings
- Fixed issue where function app settings are applied to web apps

Pipeline reference
[Web: Sites](https://github.com/Azure/ResourceModules/actions/workflows/ms.web.sites.yml)

## Type of Change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update (Wiki)

### CODE REVIEW

**Comment by MrMCake** [comment]
cc: @SeSeicht

File: arm/Microsoft.Web/sites/config/deploy.bicep [hunk 1]
```diff
@@ -2,34 +2,30 @@
...
-  'powershell'
-  ''
-])
-param functionsWorkerRuntime string = ''
+@description('Optional. Custom app settings to apply to the app')
+param customAppSettings object = {}
 
-@description('Optional. Version of the function extension.')
-param functionsExtensionVersion string = '~3'
+@description('Conditional. Required if app of kind functionapp. Resource ID of the storage account to manage triggers and logging function executions. If provided, the `AzureWebJobsStorage` setting is added to the app\'s app settings.')
```
Comment (eriqua) [inline arm/Microsoft.Web/sites/config/deploy.bicep:13]:
```suggestion
@description('Conditional. Required if the site parent resource is of kind functionapp. Resource ID of the storage account to manage triggers and logging function executions. If provided, the `AzureWebJobsStorage` setting is added to the app\'s app settings.')
```

Comment (eriqua) [reply arm/Microsoft.Web/sites/config/deploy.bicep:13]: btw is it required? no functionapp parameter file specifies it

File: arm/Microsoft.Web/sites/deploy.bicep [hunk 2]
```diff
@@ -20,30 +20,16 @@ param clientAffinityEnabled bool = true
 @description('Optional. Configuration of the app.')
 param siteConfig object = {}
 
-@description('Optional. Required if functionapp kind. The resource ID of the storage account to manage triggers and logging function executions.')
+@description('Conditional. Required if app of kind functionapp. Resource ID of the storage account to manage triggers and logging function executions. If provided, the `AzureWebJobsStorage` setting is added to the app\'s app settings.')
```
Comment (eriqua) [inline arm/Microsoft.Web/sites/deploy.bicep:23]: Also here, no reference of the storageAccountId in parameter files of kind functionapp

Comment (MrMCake) [reply arm/Microsoft.Web/sites/deploy.bicep:23]: Updated

**Comment by MrMCake** [comment]
Closing in favor of PR #1195

**Pull request closed without merging**